pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/DirectBufferedData.java

```diff
@@ -68,6 +70,47 @@ public Bytes getBytes(final long offset, final long len) {
         return Bytes.wrap(res);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getVarInt(final long offset, final boolean zigZag) {
+        return (int) getVar(Math.toIntExact(offset), zigZag);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getVarLong(final long offset, final boolean zigZag) {
+        return getVar(Math.toIntExact(offset), zigZag);
+    }
+
+    private long getVar(final int offset, final boolean zigZag) {
+        if (offset < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        int rem = Math.toIntExact(buffer.limit() - offset);
+        if (rem > 10) {
+            rem = 10;
+        }
+
+        long value = 0;
+
+        int i = 0;
+        while (i != rem) {
```

Comment:
Yeah, I agree, there are several places to read var ints that look quite similar. However, not 100% identical. There are two reasons for that. First, performance. Second, different methods have different contracts, e.g. different exceptions thrown.

I'm pretty sure it's possible to make a generic method (or two, at most, for getVarX() and readVarX() methods) and then call it from all the classes, but it would be much slower. That's why I decided to have multiple copies of this code, and each copy is slightly optimized to the corresponding class.

Reply:
Sounds good.